doc/source/reference/legacy_routines.rst

```diff
@@ -0,0 +1,24 @@
+.. _legacy_routines:
```

Comment:
@rossbar, @Mukulikaa, @rgommers - are you ok with this new top-level reference docs page for legacy API? Organizing it this way makes sense to me but want to make sure other people working on similar efforts are in sync on this.